pkg/extender/extender.go

```diff
@@ -141,15 +141,20 @@ func (e *Extender) processFilterRequest(w http.ResponseWriter, req *http.Request
 			storklog.PodLog(pod).Errorf("Error getting list of driver nodes, returning all nodes")
 		} else {
 			for _, node := range args.Nodes.Items {
-				for _, address := range node.Status.Addresses {
-					if address.Type != v1.NodeHostName {
-						continue
-					}
+				for _, driverNode := range driverNodes {
+					if e.isHostnameMatch(driverNode.ID, node.Name) &&
+						driverNode.Status == volume.NodeOnline {
+						filteredNodes = append(filteredNodes, node)
+					} else {
+						for _, address := range node.Status.Addresses {
+							if address.Type != v1.NodeHostName {
+								continue
+							}
 
-					for _, driverNode := range driverNodes {
-						if e.isHostnameMatch(driverNode.Hostname, address.Address) &&
-							driverNode.Status == volume.NodeOnline {
-							filteredNodes = append(filteredNodes, node)
+							if e.isHostnameMatch(driverNode.Hostname, address.Address) &&
+								driverNode.Status == volume.NodeOnline {
+								filteredNodes = append(filteredNodes, node)
```

Comment:
you can break out of the for loop here. No need to process subsequent `node.Status.Addresses`